openhsi/data.py

```diff
@@ -392,8 +392,11 @@ def __init__(self, n_lines:int = 16, processing_lvl:int = -1, preserve_raw:bool=
 
         self.timestamps = DateTimeBuffer(n_lines)
         self.dc_shape = (self.dc_shape[0],self.n_lines,self.dc_shape[1])
+        mem_sz = 4*reduce(lambda x,y: x*y, self.dc_shape)/2**20 # MB
+        if mem_sz > 3000 and input(f"{mem_sz:.02f} MB of RAM will be allocated. Continue? [y/n]") != "y":
+            print("Memory allocation stopped."); raise RuntimeError
+        if self.dc_shape[0] > 1: print(f"Allocated {mem_sz:.02f} MB of RAM.")
```

Comment:
Should have a flag to ignore this if desired (i.e. what happens in a remote system).

Reply:
True, I'll add that in.

Reply:
might be handy to have a method/func/util to query what a safe line count is on a platform, i.e. check memory free, make calc based on settings file?

Reply:
Will need some experimenting - it seems like this library can check available memory
https://psutil.readthedocs.io/en/latest/
I'll have a go and hopefully it works cross platform. 
Maybe this should go in a future update though and not this one?

Reply:
Thanks for the suggestion. I changed the warning to appear when trying to allocate >80% of the available RAM (can be ignored with a flag). `psutil` works cross platform so no problems there. 
This is looking good to merge to master?